### ISSUE TITLE
Error with multiple covariates in 0.17-8

### ISSUE THREAD

**Issue by tkerwin** [singmann/afex#29]
Example code:

```
require(psych)
require(afex)
data(msq)
aov_ez(data=msq, dv="Extraversion", id = "ID", between = "condition", 
    covariate=c("TOD", "MSQ_Time"), factorize=FALSE)
```

In afex 0.16-1, this works fine, but in 0.17-8, I get this error:

```
Error in `[.data.frame`(data, , between, drop = FALSE) : 
  undefined columns selected
```

It looks like the names of the covariates are being concatenated in a strange way.

**Comment by singmann** [comment]
Thanks for this report, this is most likely due to the changes in 134afd6ce24b4420fb7b00bc3d9937ed76dcea21. Paging @jonathon-love who made this changes. 
Are you willing to remove the bug you introduced (I guess)?

See:```

require(psych)
require(afex)
data(msq)
aov_ez(data=msq, dv="Extraversion", id = "ID", between = "condition", 
    covariate=c("TOD", "MSQ_Time"), factorize=FALSE, print.formula = TRUE)
# Formula send to aov_car: Extraversion ~ condition + `TOD+MSQ_Time` + Error(ID)
# Error in `[.data.frame`(data, , between, drop = FALSE) : 
#   undefined columns selected
```

**Comment by jonathon-love** [comment]
sure, i'll take a look.

cheers

**Comment by singmann** [comment]
@jonathon-love  Actually hold on. I think I got it, seems really minor.

**Comment by jonathon-love** [comment]
yup, i think it's as simple as moving line 415:

`if (!is.null(covariate)) covariate <- str_c(covariate, collapse = "+")`

down after where the covariates are escaped?

`covariate <- escape_vars(covariate)`

**Comment by singmann** [comment]
Solved in ccada51f619350293e574f7a93a165e2b5e0f5ff.

My change was similar to what you suggested:

 ```
 if (!is.null(covariate)) {
    covariate <- escape_vars(covariate)
    covariate <- str_c(covariate, collapse = "+")
  }
```
I will push a new bug-free version to CRAN latest next week. Thanks for reporting the bug.

**Comment by singmann** [comment]
Fixed version (0.18-0) is on CRAN now: https://cran.r-project.org/web/packages/afex/
